src/SqlProviderBase.ts

```diff
@@ -584,7 +590,24 @@ class SqlStoreIndex implements NoSqlProvider.DbIndex {
             checks.push(this._queryColumn + (highRangeExclusive ? ' < ' : ' <= ') + '?');
             args.push(NoSqlProviderUtils.serializeKeyToString(keyHighRange, this._keyPath));
         }
-        return this._handleQuery<T>('SELECT nsp_data FROM ' + this._tableName + ' WHERE ' + checks.join(' AND '), args, reverse, limit,
-            offset);
+        return { checks, args };
+    }
+
+    countAll(): SyncTasks.Promise<number> {
+        return this._trans.runQuery('SELECT COUNT(*) cnt FROM ' + this._tableName).then(result => result[0]['cnt']);
```

Comment:
Are you concerned with SQL injection due to "+ this._tableName"? People should not create table names directly from raw user data, but still...